src/hooks/useCreateAccount.ts

```diff
@@ -0,0 +1,103 @@
+import { EdgeAccount } from 'edge-core-js'
+
+import { completeLogin } from '../actions/LoginCompleteActions'
+import { loadTouchState } from '../actions/TouchActions'
+import { lstrings } from '../common/locales/strings'
+import { showError } from '../components/services/AirshipInstance'
+import * as Constants from '../constants/index'
+import { enableTouchId } from '../keychain'
+import { Dispatch, useDispatch } from '../types/ReduxTypes'
+import { useHandler } from './useHandler'
+import { useImports } from './useImports'
+
+export const useCreateAccount = (params: {
+  username?: string
+  password?: string
+}) => {
+  const imports = useImports()
+  const dispatch = useDispatch()
+  const {
+    context,
+    experimentConfig,
+    accountOptions,
+    onLogEvent = () => {}
+  } = imports
+
+  const isLightAccount = experimentConfig.createAccountType === 'light'
+  const handleCreateAccount = useHandler(
+    async (createAccountParams: {
+      username?: string | undefined
+      password?: string | undefined
+      pin?: string | undefined
+    }) => {
+      let error
+      try {
+        dispatch({
+          type: 'NAVIGATE',
+          data: {
+            name: 'newAccountWait',
+            params: {
+              title: lstrings.great_job,
+              message: lstrings.hang_tight + '\n' + lstrings.secure_account
+            }
+          }
+        })
+
+        const { username, password, pin } = createAccountParams
+        const account = await context.createAccount({
+          ...accountOptions,
+          username,
+          password,
+          pin
+        })
+        account.watch('loggedIn', loggedIn => {
+          if (!loggedIn) dispatch({ type: 'RESET_APP' })
+        })
+        await setTouchOtp(account, dispatch)
+
+        if (isLightAccount) {
+          dispatch(completeLogin(account))
+        } else {
+          dispatch({
+            type: 'NAVIGATE',
+            data: {
+              name: 'newAccountReview',
+              params: {
+                ...createAccountParams,
+                account
+              }
+            }
+          })
+        }
+      } catch (e: any) {
+        showError(e)
+        dispatch({
+          type: 'NAVIGATE',
+          data: {
+            name: isLightAccount ? 'newAccountPin' : 'newAccountTos',
+            params: isLightAccount ? {} : createAccountParams
+          }
+        })
+        error = String(e)
+      }
```

Comment:
It would make more sense to put the account type logic at the call-site for this API.

Reply:
Would, but we want the error attached to the `onLogEvent` below